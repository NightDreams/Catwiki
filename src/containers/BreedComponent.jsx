import React from 'react';
// import GetBreed from './GetBreed';
import useBreed from '../apiHooks/useBreed';
import Breed from '../components/organism/Breed';

export const BreedComponent = ({ breedId }) => {
  const { BreedData } = useBreed(breedId);
  if (BreedData?.loading) {
    return <h2>loading component... </h2>;
  }
  if (BreedData?.error) {
    return <h2>error component... </h2>;
  }
  return <Breed data={BreedData?.data} />;
};
