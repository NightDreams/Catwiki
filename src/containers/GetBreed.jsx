import React, { useEffect, useState } from 'react';
import Breed from '../components/organism/Breed';
import axios from 'axios';
// import useBreed from '../apiHooks/useBreed';

const GetBreed = ({ id }) => {
  // const [BreedID] = useBreed(id);
  const [BreedID, setBreedID] = useState(null);
  var rootURL = `https://api.thecatapi.com/v1/images/${id}`;

  var config = {
    headers: {
      'Content-Type': 'application/json',
      'x-api-key': '8f34cd37-dd49-45a2-a0e4-1ccd65f582b6s',
    },
  };

  useEffect(async () => {
    const { data } = await axios.get(rootURL, config);
    setBreedID(data);
  }, []);
  console.log(BreedID);
  return BreedID && <Breed data={BreedID} />;
};

export default GetBreed;
