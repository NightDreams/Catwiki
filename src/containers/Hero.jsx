import React from 'react';

import { Baner } from '../components/template/Baner';
import Galery from '../components/template/Galery';
import Post from '../components/template/Post';

const Hero = () => (
  <div className="hero">
    <Baner />
    <Galery />
    <Post />
  </div>
);
export default Hero;
