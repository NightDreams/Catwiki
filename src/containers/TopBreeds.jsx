import React, { useState } from 'react';
import Cat from '../components/template/Cat';
import useTopBreeds from '../apiHooks/useTopBreeds';

const TopBreeds = () => {
  const [page, setpage] = useState(0);
  const { top10, loading, error } = useTopBreeds(page);
  // top10 && console.log(top10);

  if (loading) {
    return <div className="TopBreeds">loading component... </div>;
  }

  if (top10) {
    return (
      <div className="TopBreeds">
        <h2>Top 10 most searched breeds</h2>
        <button onClick={() => setpage(page + 1)}>next </button>
        <div className="catList">
          {top10.map((e) => (
            <Cat
              key={e.id}
              src={e.image.url}
              name={e.name}
              desc={e.description}
              imgID={e.image.id}
            />
          ))}
        </div>
      </div>
    );
  }
  if (error) {
    return <div className="TopBreeds">error component... </div>;
  }
};

export default TopBreeds;
