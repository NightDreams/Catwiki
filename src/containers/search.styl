@media screen and (min-width: 1440px) {
  .search {
    padding: unset;
    // position: absolute;
    margin-top: 52px;
    padding: unset;
  }

  .baner .search {
    padding: unset;
  }

  .search input {
    border: none;
    width: 394.62px;
    font: 500 18px / 22px Montserrat;
    height: 100%;

    // background: #FFFFFF;
    // border-radius: 59px;
    // padding: 0 28px;
    // height: 69.67px;
    &:focus {
      outline: none;
    }
  }

  // scrollbar
  .SearchList::-webkit-scrollbar {
    // background: #F2F2F2;
    // border-radius: 12px;
    width: 10px;
  }

  .SearchList::-webkit-scrollbar-thumb {
    border-radius: 10px;
    height: 50px;
    background: #BDBDBD;
  }

  #on {
    display: block;
    padding: 0;
    color: black;
    position: absolute;
  }

  .search .searchWrapper {
    padding: 0 28px;
  }

  .searchWrapper {
    background: #fff;
    border: none;
    border-radius: 59px;
    padding: 0 28px;
    width: 394.62px;
    height: 69.67px;
    font: 500 18px / 22px Montserrat;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .SearchList {
    width: 394.62px;
    height: 219.72px;
    overflow: auto;
    border-radius: 24px;
    background: white;
    margin-top: 16px;
    display: none;
    scrollbar-width: thin;

    & p {
      font: 500 18px / 22px Montserrat;
      width: 360.01px;
      height: 56.16px;
      display: flex;
      align-items: center;
      border-radius: 12px;
      padding: 0 28px;
    }

    & p:hover {
      background: rgba(151, 151, 151, 0.1);
    }
  }
}