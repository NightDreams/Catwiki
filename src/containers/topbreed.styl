@media screen and (min-width: 1440px) {
  /* Elemento | http://localhost:8097/top10 */
  .layout > div:nth-child(2) > div:nth-child(1) > h2:nth-child(1) {
    font: 700 48px / 58px Montserrat;
  }

  /* Elemento | http://localhost:8097/top10 */
  .cat > img:nth-child(1) {
    width: 170px;
    height: 170px;
    border-radius: 24px;
    outline: 1px solid;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  /* Elemento | http://localhost:8097/top10 */
  .cat > h3:nth-child(2) {
    font: 600 36px / 43px Montserrat;
  }

  /* Elemento | http://localhost:8097/top10 */
  .cat > p:nth-child(3) {
    font: var(--parrafo-p);
  }
}