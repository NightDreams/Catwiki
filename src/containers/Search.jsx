import React, { useState, useRef, useMemo, useEffect } from 'react';
import { Link } from '@reach/router';
import useBreedList from '../apiHooks/useBreedList';
import './search.styl';
import { MdSearch } from 'react-icons/md';
import { IconContext } from 'react-icons';

const Search = () => {
  const { BreedList } = useBreedList();
  const [search, setSearch] = useState('');
  const [on, seton] = useState(null);

  const searchInput = useRef(null);
  const handleSearch = () => {
    setSearch(searchInput.current.value);
  };
  const filterBreeds = useMemo(
    () =>
      BreedList?.data.filter((breed) => {
        return breed.name.toLowerCase().includes(search.toLowerCase());
      }),
    [BreedList.data, search]
  );
  if (BreedList?.loading) {
    return <div className="TopBreeds">loading component... </div>;
  }
  if (BreedList?.error) {
    return <div className="TopBreeds">error component... </div>;
  }

  return (
    <div className="search">
      <div className="searchWrapper">
        <input
          ref={searchInput}
          type="text"
          value={search}
          onChange={handleSearch}
          placeholder="Search"
          onClick={() => (!on ? seton('on') : seton(''))}
        />

        <IconContext.Provider
          value={{
            size: '2rem',
            color: 'black',
          }}
        >
          <MdSearch />
        </IconContext.Provider>
      </div>

      <div className={`SearchList`} id={on}>
        {filterBreeds.map((e) => (
          <Link key={e.id} to={`/breed/${e.reference_image_id}`}>
            <p>{e.name}</p>
          </Link>
        ))}
        {/* {BreedList.map((e) => (
            <p key={e.id}>{e.name}</p>
          ))} */}
      </div>
    </div>
  );
};

export default Search;
