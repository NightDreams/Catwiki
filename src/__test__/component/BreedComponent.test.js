import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import { BreedComponent } from '../../containers/BreedComponent';

describe('<BreedComponent>', () => {
  const comp = mount(<BreedComponent />);
  test('Render <BreedComponent>', () => {
    expect(comp.length).toEqual(1);
  });
  test('Render Breed ', () => {
    expect(comp.find('.BreedData'));
  });
});

describe('<BreedComponent> snapshot', () => {
  const comp = create(<BreedComponent />);
  expect(comp.toJSON()).toMatchSnapshot();
});
