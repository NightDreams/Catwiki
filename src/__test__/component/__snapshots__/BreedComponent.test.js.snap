// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<h2>
  loading component... 
</h2>
`;
